src/subject_name/verify.rs

```diff
@@ -345,20 +345,9 @@ impl<'a> Iterator for NameIterator<'a> {
             }
         }
 
-        if let Some(subject_directory_name) = self.subject_directory_name.take() {
-            return Some(Ok(GeneralName::DirectoryName(subject_directory_name)));
-        }
-
-        if let Some(subject_common_name) = self.subject_common_name.take() {
-            return match common_name(subject_common_name) {
-                Ok(Some(cn)) => Some(Ok(GeneralName::DnsName(cn))),
-                Ok(None) => None,
-                // All the iterator fields should be `None` at this point
-                Err(err) => Some(Err(err)),
-            };
-        }
-
-        None
+        self.subject_directory_name
```

Comment:
I think this is just reformulating the `if let Some()` that was there before, right? I wrote that, and I still feel like it was easier to comprehend than this combinator-based spelling.

(If we want to change it, I'd prefer to have that split in a separate commit from the logic removing the `self.subject_common_name` handling.)